planner/core/logical_plan_builder.go

```diff
@@ -5875,11 +5875,44 @@ func buildWindowSpecs(specs []ast.WindowSpec) (map[string]*ast.WindowSpec, error
 	return specsMap, nil
 }
 
+func unfoldSelectList(list *ast.SetOprSelectList, unfoldList *ast.SetOprSelectList) {
+	for _, sel := range list.Selects {
+		switch s := sel.(type) {
+		case *ast.SelectStmt:
+			unfoldList.Selects = append(unfoldList.Selects, s)
+		case *ast.SetOprSelectList:
+			unfoldSelectList(s, unfoldList)
+		}
+	}
+}
+
 // extractTableList extracts all the TableNames from node.
 // If asName is true, extract AsName prior to OrigName.
 // Privilege check should use OrigName, while expression may use AsName.
+// TODO: extracting all tables by vistor model maybe a better way
 func extractTableList(node ast.ResultSetNode, input []*ast.TableName, asName bool) []*ast.TableName {
```

Comment:
Maybe we can add a unit test for `extractTableList` ?

Reply:
That's a big project. Most internal methods in `logic_plan_builder.go` do not have a UT, such as `extractTableSourceAsNames` , `extractOnCondition` .... and so on.

Reply:
Adding a unit test is a better practice to prevent explosion of combinations. So even we have some integration tests we still need unit tests to ensure robustness of our code. But I will approve the pr because it is for `forbidden bind sql when subquery or union exists` . We may add a unit test for this method in another RP.

Reply:
I agree to make a specific project to improve the test coverage about `logic_plan_builder.go`